### PR TITLE
Fix release 1.20 README self link

### PR DESCRIPTION
**Pull request by ebriand**
Hello 👋 

Just a small PR to fix release 1.20 README self link 😄 

#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:
Fix release 2.20 README.md self link

### CODE REVIEW

File: releases/release-1.20/README.md [hunk 1]
```diff
@@ -105,4 +105,4 @@ Please refer to the [release phases document](../release_phases.md).
 [1.20-blocking]: https://testgrid.k8s.io/sig-release-1.20-blocking#Summary
 
 [exception requests]: ../EXCEPTIONS.md
-[release phases document]: ../release_phases.md
\ No newline at end of file
+[release phases document]: ../release_phases.md
```
Comment (hasheddan) [inline releases/release-1.20/README.md]: Is this change just a newline removal?

Comment (ebriand) [reply releases/release-1.20/README.md]: Hi @hasheddan 👋  I directly edited the file in Github interface, seems it has done that by itself 🤔  I will fix that 👍

Comment (hasheddan) [reply releases/release-1.20/README.md]: awesome! thanks @ebriand :)

Comment (ebriand) [reply releases/release-1.20/README.md]: Done 👌  Thanks @hasheddan 👍

**Review by hasheddan — APPROVED** [review]
Thanks @ebriand!

/lgtm

**Pull request merged**